### ISSUE TITLE
Feature request: Improve the version info for the cores

### ISSUE THREAD

**Issue by davidhedlund** [libretro/RetroArch#16684]
## Description

`Main Menu -> Information -> System Information` has:
* RetroArch Version
* Git Version
* Build Date

Example:
![image](https://github.com/libretro/RetroArch/assets/6416073/da2b6352-1277-4b29-b6be-737a829ce5c6)

### Expected behavior

* Add "Core Version" -- this is useful to make it simple to follow up core bug reports
* Replace "Core Version" with "Git Version"?
* Add "Build Date" -- this is useful to make it simple to follow up core bug reports

### Actual behavior

![image](https://github.com/libretro/RetroArch/assets/6416073/25205ac1-a132-4b4e-b84e-a5162511501a)

### Steps to reproduce the bug

* `Main Menu -> Load Core -> [Any core]`
* `Main Menu -> Information -> Core Information`

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.19.0

### Environment information

- OS: GNU/Linux

**Comment by RobLoach** [comment]
> Add "Core Version" -- this is useful to make it simple to follow up core bug reports

This is already available in Core Information. Are you proposing moving the Core Information menu to the System Information? Because they accomplish different things.

> Replace "Core Version" with "Git Version"?

The "Core Version" is not Git Version, it's `retro_system_info`'s `library_version`. The `Makefile`s usually just append the git commit to allow verification of versions.

> Add "Build Date" -- this is useful to make it simple to follow up core bug reports

While it would be useful, build dates are not currently bundled in by the cores, nor in the buildbot. I also don't think it should do this as that can be found via the git commit hash. Go to the PCSX-ReArmed git repo, and find `9aefd42`.

**Comment by davidhedlund** [comment]
Thank you for your feedback!

> > Add "Core Version" -- this is useful to make it simple to follow up core bug reports
> 
> This is already available in Core Information. Are you proposing moving the Core Information menu to the System Information? Because they accomplish different things.

The current "Core Version" is "Core Library Version". So, I suggest that we rename the current "Core Version" to "Core Library Version". (see below). Then we add "Core Version" that is displaying conventional version number like ("RetroArch Version: 1.19.0") but for the core.

> 
> > Replace "Core Version" with "Git Version"?
> 
> The "Core Version" is not Git Version, it's `retro_system_info`'s `library_version`. The `Makefile`s usually just append the git commit to allow verification of versions.

Sorry, let me rephrase: Replace "Core Version" with "Core Library Version"

> 
> > Add "Build Date" -- this is useful to make it simple to follow up core bug reports
> 
> While it would be useful, build dates are not currently bundled in by the cores, nor in the buildbot.

Thank you.

> I also don't think it should do this as that can be found via the git commit hash. Go to the PCSX-ReArmed git repo, and find `9aefd42`.

While it's possible and routinely made by developers, it's inconvenient for end users to do that. It would be useful to find out if a core is outdated by end users by quickly viewing `Main Menu -> Information -> Core Information` for example

# Summary example:
* Core Name: PCSX-ReArmed
* Core Version:  [1.0 for example]
* Core Library Version [renamed from "Core Version"]: r24l 9aefd42
* Build Date: [YYYY-MM-DD]
* Git Version: [whatever]

I recommend populating the retro_system_info data files with entries that are currently missing (the square bracket values in the list above).

**Comment by davidhedlund** [comment]
@zoltanvb Hey! For starters, it should be simple and useful to add this to `Main Menu -> Information -> Core Information`:  
* Updated: [YYYY-MM-DD]  

That way, users can easily see when the core was last updated on their system.  
Feels like a neat little touch to make things clearer. What do you think?

**Comment by davidhedlund** [comment]
* https://github.com/libretro/RetroArch/issues/18299
